visualize-admin/visualization-tool
# fix: Allow multiple identical labels in charts

@bprusinowski is the author of the PR.

Fixes #1015.

This PR changes the way we index observations from indexing by label (or abbreviation) to indexing by value (for hierarchical dimensions, passed in a form of an additional property: `dimensionIri/__iri__`) or label (or abbreviation).

Basically, we change
`{ label_or_abbreviation: Observation }` into
`{ value_or_label_or_abbreviation: Observation }`.

We needed to make this change to allow displaying values with the same labels, which is now a case with hierarchical dimensions (but in most places, we are able to distinguish them anyway, either by a color legend below a chart or visually on the map).

There are also smaller bug fixes / improvements included:
- showing color legend when it makes sense for line chart (when there is more than one value)
- showing map legend when number of values > 1
- mouse interactions with overlapping shapes (map)
- added text stroke to `CircleLegend` labels
- added `dataSet` prop to `PublishedConfiguratorState` to fix loading hierarchies for `ColorLegend`

### How to test
1. Go to https://visualization-tool-git-fix-same-labelled-ixt1.vercel.app/en/v/QIhTScAJpRyN?dataSource=Int
2. See that we show Jura twice.
3. Copy and edit this visualization.
4. Change chart type to map.
5. See that both Jura regions are displayed on the map (and that they are colored correctly and that the tooltip interactions work).
6. Compare with https://int.visualize.admin.ch/en/v/eDRM5d2-KeD6?dataSource=Int
7. Follow steps 2-5 and see that the charts and interactions are broken.
